src/main/c/Makefile

```diff
@@ -1,290 +1,297 @@
+arm64v8:
+	$(MAKE) -f natives.mk
+
+# Requires an Android NDK be installed, and its tools be on your PATH.
+#
+# This rule has not been tested.
+android: export CC := arm-linux-androideabi-gcc
+android: export CFLAGS += $(LINUX_CFLAGS)
+android: export platform := linux/ARM_ANDROID
+android:
+	$(MAKE) -f natives.mk
+
+# Requires gcc-powerpc-linux-gnu.
+ppc: export CC := powerpc-linux-gnu-gcc
+ppc: export CFLAGS += $(LINUX_CFLAGS)
+ppc: export platform := linux/PPC
+ppc:
+	$(MAKE) -f natives.mk
+
+# Requires a macOS host.
+osx: export CFLAGS += -I$(JAVA_HOME)/include/darwin -arch x86_64
+osx: export LDFLAGS := -arch x86_64 -dynamiclib -framework JavaVM -framework IOKit -framework CoreFoundation
+osx: export objects := fuserImp.o SerialImp.o
+osx: export platform := osx
+osx: export lib_type := jnilib
+osx:
+	$(MAKE) -f natives.mk
+
+freebsd: freebsd32 freebsd64
+
+# Requires a FreeBSD host.
+freebsd32: export CFLAGS += -I$(JAVA_HOME)/include/freebsd -m32
+freebsd32: export LDFLAGS += -m32
+freebsd32: export platform := freebsd/x86_32
+freebsd32:
+	$(MAKE) -f natives.mk
+
+# Requires a FreeBSD host.
+freebsd64: export CFLAGS += -I$(JAVA_HOME)/include/freebsd -m64
+freebsd64: export LDFLAGS += -m64
+freebsd64: export platform := freebsd/x86_64
+freebsd64:
+	$(MAKE) -f natives.mk
+
+windows: windows32 windows64
+# Don't embed the build timestamp into DLLs so the output is the same from one
+# compile to the next for the same input source.
+# https://blog.beuc.net/posts/Practical_basics_of_reproducible_builds_2/
+WINDOWS_LDFLAGS := -Wl,--add-stdcall-alias -Wl,--no-insert-timestamp
+
+# Requires gcc-mingw-w64-i686 (included in gcc-mingw-w64). On Windows, this,
+# along with mingw32-make, is most easily acquired in the form of TDM-GCC:
+# https://jmeubank.github.io/tdm-gcc/.
+ifeq ($(OS),Windows_NT)
+# On Windows, TDM-GCC includes one GCC binary which can produce both 32- and
+# 64-bit binaries.
+windows32: export CC := x86_64-w64-mingw32-gcc
+else
+# Elsewhere, Mingw-w64 ships a separate GCC frontend for each target
+# architecture.
+windows32: export CC := i686-w64-mingw32-gcc
+endif
+# We include the only relevant Windows-specific Java header in the repository
+# and compile against that path, instead of $(JAVA_HOME)/include/win32, because
+# we can cross-compile the Windows natives from other platforms via Mingw-w64,
+# and other platform JDK installations will only have their own headers.
+windows32: export CFLAGS += -I./include/windows -I./include/windows/win32 -m32
+windows32: export LDFLAGS += $(WINDOWS_LDFLAGS) -m32
+windows32: export objects += init.o termios.o
+windows32: export platform := windows/x86_32
+windows32: export lib_type := dll
+windows32:
+	$(MAKE) -f natives.mk
+
+# Requires gcc-mingw-w64-x86-64 (included in gcc-mingw-w64).
+windows64: export CC := x86_64-w64-mingw32-gcc
+windows64: export CFLAGS += -I./include/windows -I./include/windows/win32 -m64
+windows64: export LDFLAGS += $(WINDOWS_LDFLAGS) -m64
```

Comment:
In my environment, the linked DLL has a dynamic dependency to libgcc_s_seh-1.dll. That causes the loading in the JVM to fail. I needed to add a `static` parameter to make it work.
```
$ x86_64-w64-mingw32-gcc -v
Using built-in specs.
COLLECT_GCC=x86_64-w64-mingw32-gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-w64-mingw32/7.3-win32/lto-wrapper
Target: x86_64-w64-mingw32
Configured with: ../../src/configure --build=x86_64-linux-gnu --prefix=/usr --includedir='/usr/include' --mandir='/usr/share/man' --infodir='/usr/share/info' --sysconfdir=/etc --localstatedir=/var --disable-silent-rules --libdir='/usr/lib/x86_64-linux-gnu' --libexecdir='/usr/lib/x86_64-linux-gnu' --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --enable-shared --enable-static --disable-multilib --with-system-zlib --libexecdir=/usr/lib --without-included-gettext --libdir=/usr/lib --enable-libstdcxx-time=yes --with-tune=generic --with-headers=/usr/x86_64-w64-mingw32/include --enable-version-specific-runtime-libs --enable-fully-dynamic-string --enable-libgomp --enable-languages=c,c++,fortran,objc,obj-c++,ada --enable-lto --with-plugin-ld --enable-threads=win32 --program-suffix=-win32 --program-prefix=x86_64-w64-mingw32- --target=x86_64-w64-mingw32 --with-as=/usr/bin/x86_64-w64-mingw32-as --with-ld=/usr/bin/x86_64-w64-mingw32-ld --enable-libatomic --enable-libstdcxx-filesystem-ts=yes
Thread model: win32
gcc version 7.3-win32 20180312 (GCC)
```
```suggestion
windows64: export LDFLAGS += $(WINDOWS_LDFLAGS) -m64 -static-libgcc
```

Reply:
Ah, interesting! The existing Makefile [links with `-static-libgcc -static-libstdc++` on Windows](https://github.com/NeuronRobotics/nrjavaserial/blob/40587564401997d31c1f2a73b20a14cde2f04cf7/src/main/c/Makefile.ubuntu64#L44), and I clearly forgot to hunt down why. It worked on my machine, because I have [TDM-GCC](https://jmeubank.github.io/tdm-gcc/)'s `bin/` directory on my `PATH`, and it contains a copy of that library. D'oh!

You're finding that the library works fine with just `-static-libgcc` and without `-static-libstdc++`? I can't imagine the library needs anything from libstdc++, as it's compiled and linked with a C compiler.

Reply:
`-static-libstdc++` is not necessary. Just checked `windows32`, `-static-libgcc` is there necessary, too.

Reply:
> Just checked `windows32`, `-static-libgcc` is there necessary, too.

Ah, sorry, I should've mentioned that I found that out myself by checking the most recent build artifacts from this branch with [Dependency Walker](https://dependencywalker.com/). Yeah, I'll add `-static-libgcc` to `WINDOWS_LDFLAGS`.